converter/ena/ena_run.py

```diff
@@ -2,32 +2,47 @@
 from config import ENA_FTP_DIR
 from converter.ena.classes import Run, RunType, FileFiletype, RunSet, AttributeType
 from converter.ena.classes.sra_common import RefObjectType
-from converter.ena.base import EnaModel
+from converter.ena.base import EnaModel, XMLType, EnaArchiveException
+from converter.ena.ena_receipt import EnaReceipt
 
 
 class EnaRun(EnaModel):
 
     FILE_CHECKSUM_METHOD = 'MD5'
     SEQUENCE_FILE_TYPES = ['fq', 'fastq', 'fq.gz', 'fastq.gz']
 
-    def __init__(self, hca_data):
-        self.logger = logging.getLogger(__name__)
-        self.submission = hca_data.submission
-        self.assays = hca_data.submission["assays"]
-
-        self.ena_upload_area_path = f'{ENA_FTP_DIR}/{self.submission["uuid"]["uuid"]}/'
-
-    def create_set(self):
+    def __init__(self, experiment_ref):
+        self.experiment_ref = experiment_ref
+
+    # Todo: extract function to parent class
+    def archive(self, assay):
+        run = self.create(assay)
+        input_xml = self.xml_str(run)
+        receipt_xml = self.post(XMLType.RUN, input_xml, update=True if run.accession else False)
+        accessions = EnaReceipt(XMLType.RUN, input_xml, receipt_xml).process_receipt()
+        if accessions and len(accessions) == 1:
+            return accessions[0]
+        raise EnaArchiveException('Ena archive no accession returned.')
+
+
+    def archive(self, assay):
```

Comment:
It looks like you are redefining the `archive` function here.

Reply:
fixed, amnon also pointed this out.